### PR TITLE
Set Volatility of VOLATILE system function to STABLE

### PR DESCRIPTION
**Pull request by skkyal**
Signed-off-by: Shlok Kumar Kyal (skkyal@amazon.com)

PR in 3.x: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/1109

### Description

Set Volatility of VOLATILE system function to STABLE 

There are 1053 system functions in the sys schema, of them 331 are volatile. All the VOLATILE system functions should be set to STABLE as this is a critical basis to enable index usage in queries (other conditions are needed as well, but function stability is a must).
In this PR made all system functions as STABLE, except the functions whose test cases were missing or do not satisfy the criteria to be STABLE. Of 331 Volatile system functions, 194 functions were made STABLE.

Count of Volatile Functions Before Change
```
1> select count(*) from pg_proc where provolatile = 'v' AND pronamespace = 16398 AND prokind = 'f'
2> go
count
-----
331
```

Count of Volatile Functions After Change
```
1> select count(*) from pg_proc where provolatile = 'v' AND pronamespace = 16398 AND prokind = 'f'
2> go
count
-----
137
```
So made 194 system functions as STABLE

### Issues Resolved

BABEL-3790

### Test Scenarios Covered ###
Made only that system functions as STABLE which were already having tests cases
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by KushaalShroff** [comment]
Can you please add more information to description on the count of functions made stable and how many were not.

Also for the purpose of review, can you please grep your "diff" to get me the number of "STABLE" occurrences in fresh-install scripts and number of occurrences of "Create or replace" in upgrade scripts. (Make sure both are equal and add the output as a reply to this comment)

**Comment by skkyal** [comment]
> Can you please add more information to description on the count of functions made stable and how many were not.
> 
> Also for the purpose of review, can you please grep your "diff" to get me the number of "STABLE" occurrences in fresh-install scripts and number of occurrences of "Create or replace" in upgrade scripts. (Make sure both are equal and add the output as a reply to this comment)

checking with grep was getting tricky, so checked after doing upgrade locally
fresh build
```
1> select count(*) from pg_proc where provolatile = 'v' AND pronamespace = 16398 AND prokind = 'f'
2> go
count
-----
137
```
after upgrade changes
```
1> select count(*) from pg_proc where provolatile = 'v' AND pronamespace = 16398 AND prokind = 'f'
2> go
count
-----
137
```
Total 331 system functions were there initially. Made 194 functions as STABLE

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql [hunk 1]
```diff
@@ -162,6 +162,9330 @@ WHERE has_schema_privilege(s.schema_id, 'USAGE')
 AND c.contype = 'c' and c.conrelid != 0;
 GRANT SELECT ON sys.check_constraints TO PUBLIC;
 
+-- make sys functions stable
+CREATE OR REPLACE FUNCTION sys.schema_id()
```
Comment (kuntalghosh) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql]: Use ALTER FUNCTION statement to change VOLATILE function to STABLE. For example,
```
ALTER FUNCTION sys.schema_id() STABLE;
```

Redefining so many functions is error-prone and the git commit context of individual functions will be lost.

**Review by kuntalghosh — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql (hunk 1 above)
Comment (skkyal) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.3.0--2.4.0.sql]: Have done the above changes

**Review by KushaalShroff — DISMISSED** [review]

**Review by kuntalghosh — DISMISSED** [review]

**Pull request merged**